homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,221 @@
+            vol.Optional(
+                CONF_STILL_IMAGE_URL,
+                description={
+                    "suggested_value": user_input.get(CONF_STILL_IMAGE_URL, "")
+                },
+            ): str,
+            vol.Optional(
+                CONF_STREAM_SOURCE,
+                description={"suggested_value": user_input.get(CONF_STREAM_SOURCE, "")},
+            ): str,
+            vol.Optional(
+                CONF_RTSP_TRANSPORT, default=user_input.get(CONF_RTSP_TRANSPORT)
+            ): vol.In([None, "tcp", "udp", "udp_multicast", "http"]),
+            vol.Optional(
+                CONF_AUTHENTICATION, default=user_input.get(CONF_AUTHENTICATION)
+            ): vol.In([HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION]),
+            vol.Optional(
+                CONF_USERNAME,
+                description={"suggested_value": user_input.get(CONF_USERNAME, "")},
+            ): str,
+            vol.Optional(
+                CONF_PASSWORD,
+                description={"suggested_value": user_input.get(CONF_PASSWORD, "")},
+            ): str,
+            vol.Optional(
+                CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+                default=user_input.get(CONF_LIMIT_REFETCH_TO_URL_CHANGE),
+            ): bool,
+            vol.Optional(
+                CONF_CONTENT_TYPE, default=user_input.get(CONF_CONTENT_TYPE)
+            ): str,
+            vol.Optional(
+                CONF_FRAMERATE,
+                description={"suggested_value": user_input.get(CONF_FRAMERATE, "")},
+            ): int,
+            vol.Optional(
+                CONF_VERIFY_SSL, default=user_input.get(CONF_VERIFY_SSL)
+            ): bool,
+        }
+        return vol.Schema(spec)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the start of the config flow."""
+        errors = {}
+        if user_input is not None:
+            # Secondary validation because serialised vol can't seem to handle this complexity:
+            if user_input.get(CONF_STILL_IMAGE_URL) in [None, ""] and user_input.get(
+                CONF_STREAM_SOURCE
+            ) in [None, ""]:
+                errors["base"] = "no_still_image_or_stream_url"
+            else:
+                (res, errors["base"]) = await self._test_connection(user_input)
+                if res:
+                    return self.async_create_entry(
+                        title=user_input[CONF_NAME], data=user_input
+                    )
+        else:
+            user_input = DEFAULT_DATA.copy()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.build_schema(user_input),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_config):
+        """Handle config import from yaml."""
+        # abort if we've already got this one.
+        self._async_abort_entries_match(
```

Comment:
We should check `CONF_STREAM_SOURCE` as well

Reply:
Agreed, done.